crawl-ref/source/main.cc

```diff
@@ -560,16 +560,52 @@ static void _show_commandline_options_help()
 #endif
 }
 
-static void _wanderer_note_items()
+// Announce to the message log and make a note of the player's starting items,
+// spells and spell library
+static void _wanderer_note_equipment()
 {
     const string equip_str =
-        you.your_name + " set off with: "
+        "the following items: "
         + comma_separated_fn(begin(you.inv), end(you.inv),
                              [] (const item_def &item) -> string
                              {
                                  return item.name(DESC_A, false, true);
                              }, ", ", ", ", mem_fn(&item_def::defined));
-    take_note(Note(NOTE_MESSAGE, 0, 0, equip_str));
+
+    // Wanderers start with at most 1 spell memorised.
+    const string spell_str =
+        !you.spell_no ? "" :
+        "; and the following spell memorised: "
+        + comma_separated_fn(begin(you.spells), end(you.spells),
+                             [] (const spell_type spell) -> string
+                             {
```

Comment:
Perhaps this `comma_separated_fn` can be simplified to just `you.spells[0]` if wanderers start with at most 1 level 1 spell memorised? Should there also be an assert somewhere that you.spell_no = 0 or 1?